ppa/archive/templates/archive/snippets/cluster_work.html

```diff
@@ -1,5 +1,5 @@
 {# NOTE: expects cluster_id to be passed in #}
 <div class="cluster-work icon container" >
     {# TODO: include count if we can easily get it ; search and browse x digitized works within cluster #}
-    <a href="{% url "archive:list" %}?cluster={{ cluster_id }}">search and browse within cluster</a>
+    <a href="{% url "archive:list" %}?cluster={{ cluster_id }}{% if query %}&query={{ query }}{% endif %}">search and browse within cluster</a>
```

Comment:
Here's what I suggest; I think this should preserve all query/form args, but make sure that the cluster search starts on page 1 of the results — you'll want to check, of course.

```suggestion
    <a href="{% url "archive:list" %}?cluster={{ cluster_id }}&?{% querystring_replace page=1 %}">search and browse within cluster</a>
```

Reply:
I see-- nice-- we already have a querystring_replace function which takes a page argument. I just added this and deployed, seems to be working.